qemu/tests/openflow_acl_test.py

```diff
@@ -277,11 +277,27 @@ def setup_service(setup_target):
         error_context.context("Set up %s service in %s"
                               % (setup_params.get("service"), setup_target),
                               logging.info)
-        setup_func(setup_cmd, timeout=setup_timeout)
         if params.get("copy_ftp_site") and setup_target != "localhost":
-            ftp_site = os.path.join(data_dir.get_deps_dir(), params.get("copy_ftp_site"))
+            ftp_site = os.path.join(data_dir.get_deps_dir(),
+                                    params.get("copy_ftp_site"))
             ftp_dir = params.get("ftp_dir")
             setup_vm.copy_files_to(ftp_site, ftp_dir)
+        access_param = setup_params.object_params(setup_target)
+        if "ftp" in access_param.get("access_cmd"):
+            setup_func("rm -f %s" % params["vsftpd_conf_tmp"])
+            vsftpd_content = setup_func("cat %s" %
+                                        params["vsftpd_conf"]).strip()
+            lines = vsftpd_content.split("\n")
+            for i in range(len(lines)):
+                if "anonymous_enable=NO" not in lines[i]:
+                    setup_func("echo \"%s\" >> "
+                               "%s" % (lines[i], params["vsftpd_conf_tmp"]))
+                else:
+                    setup_func("echo \"anonymous_enable=YES\" >> %s" %
+                               params["vsftpd_conf_tmp"])
```

Comment:
Seems the purpose is to replace `anonymous_enable=NO` with `anonymous_enable=YES`, why not directly send `(sed -i 's/anonymous_enable=NO/anonymous_enable=YES/g %s) % params["vsftpd_conf"]`, which is less time and space cost?

Reply:
@vivianQizhu Updated.